CHANGELOG.md

```diff
@@ -15,6 +15,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
   The package contains semantic conventions from the `v1.31.0` version of the OpenTelemetry Semantic Conventions.
   See the [migration documentation](./semconv/v1.31.0/MIGRATION.md) for information on how to upgrade from `go.opentelemetry.io/otel/semconv/v1.30.0`(#6479)
 - Add `Recording`, `Scope`, and `Record` types in `go.opentelemetry.io/otel/log/logtest`. (#6507)
+- Reduce memory usage in `FixedSizeReservoir` by switching to `math/rand/v2` functions instead of allocating a rand.Source. (#6693)
```

Comment:
```suggestion
- Reduce memory usage in `FixedSizeReservoir` by switching to `math/rand/v2` functions so that allocating `rand.Source` is no longer necessary. (#6693)
```